publisher/server/publisher.go

```diff
@@ -113,12 +115,29 @@ func (s *Server) IsPublishableApp(name string) bool {
 		log.Println(err)
 		return false
 	}
-	if version >= latestRunningVersion(s.EtcdClient, appName) {
+
+	if version >= latestRunningVersion(s.EtcdClient, appName) && isPortOpen(hostAndPort) {
 		return true
 	}
 	return false
 }
 
+func isPortOpen(hostAndPort string) bool {
+	portOpen := false
+	if hostAndPort == "localhost:-1" {
+		portOpen = true
+	} else if hostAndPort == "localhost:-2" {
+		portOpen = false
+	} else {
+		conn, err := net.Dial("tcp", hostAndPort)
+		if err != nil {
```

Comment:
Correct. I re-wrote the tests following the paradigm of https://golang.org/src/net/dial_test.go  to avoid such errors.